CMakeLists.txt

```diff
@@ -185,7 +185,7 @@ IF(UNIX)
   # ensure compatibility with older CPUs
   # enable C++11 features
   add_definitions(-DLINUX_BUILD)
-  SET(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -g -Wall -Wno-unused-variable")
+  SET(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -g -Wall -Wno-unused-variable -DDFHACK_ASSERT")
```

Comment:
Also, maybe add_definitions() would work (if that can apply to non-Release builds).